addons/hr/models/hr_department.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, fields, models, _
+from openerp.exceptions import ValidationError
+
+
+class HrDepartment(models.Model):
+
+    _name = "hr.department"
+    _description = "HR Department"
+    _inherit = ['mail.thread', 'ir.needaction_mixin']
+
+    name = fields.Char(string='Department Name', required=True)
+    company_id = fields.Many2one('res.company', string='Company', index=True, required=False, default=lambda self: self.env.user.company_id)
+    parent_id = fields.Many2one('hr.department', string='Parent Department', index=True)
+    child_ids = fields.One2many('hr.department', 'parent_id', string='Child Departments')
+    manager_id = fields.Many2one('hr.employee', string='Manager', track_visibility='onchange')
+    member_ids = fields.One2many('hr.employee', 'department_id', string='Members', readonly=True)
+    jobs_ids = fields.One2many('hr.job', 'department_id', string='Jobs')
+    note = fields.Text()
+    color = fields.Integer('Color Index')
+
+    @api.multi
+    def name_get(self):
+        result = []
+        for department in self:
+            name = department.name
+            if department.parent_id:
+                name = department.parent_id.name + ' / ' + name
+            result.append((department.id, name))
+        return result
```

Comment:
@WHPODOO 

Without name_get method, display_name field won't display full name in tree view of hr_department.